api/dbadapters/bigquery.ts

```diff
@@ -190,6 +194,28 @@ export class BigQueryDbAdapter implements IDbAdapter {
 
   public async close() {}
 
+  public async persistedStateMetadata(
+    compiledGraph: dataform.ICompiledGraph
+  ): Promise<dataform.IPersistedTableMetadata[]> {
+    const adapter = adapters.create(
+      compiledGraph.projectConfig,
+      compiledGraph.dataformCoreVersion || "1.0.0"
+    );
+    try {
+      const { rows } = await this.runQuery(
+        `SELECT * FROM ${adapter.resolveTarget(STATE_PERSIST_TABLE_TARGET)}`,
+        5000 // not expecting to have more dataset than this
```

Comment:
this is definitely a bit concerning, as we have seen some users with insane numbers of tables, especially with a few users attached to a large project which has IDS turned on.

i wonder if this suggests we should be rethinking the metadata table's schema, to have separate columns for database/schema/table, so that we can do smaller lookups.

in any case, there should be a check here that the returned number of rows is not equal to 5000 - and if it is, throw an Error or something as at that point we can't reliably return results

Reply:
Added comment